backend/src/main/java/ai/verta/modeldb/versioning/blob/diff/ConflictGenerator.java

```diff
@@ -385,20 +497,29 @@ private static AutogenConfigDiff getConfigConflictBlob(
     return retMap;
   }
 
-  private static AutogenCodeDiff getCodeConflictBlob(
+  private static List<AutogenCodeDiff> getCodeConflictBlob(
       AutogenCodeDiff a, AutogenCodeDiff b, AutogenCodeBlob c) {
-    return Utils.removeEmpty(
-        new AutogenCodeDiff()
-            .setGit(
-                getGitDiff(
-                    a == null ? null : a.getGit(),
-                    b == null ? null : b.getGit(),
-                    c == null ? null : c.getGit()))
-            .setNotebook(
-                getNoteBookDiff(
-                    a == null ? null : a.getNotebook(),
-                    b == null ? null : b.getNotebook(),
-                    c == null ? null : c.getNotebook())));
+    AutogenGitCodeDiff gitConflict =
+        getGitDiff(
+            a == null ? null : a.getGit(),
+            b == null ? null : b.getGit(),
+            c == null ? null : c.getGit());
+
+    AutogenNotebookCodeDiff notebookConflict =
+        getNoteBookDiff(
+            a == null ? null : a.getNotebook(),
+            b == null ? null : b.getNotebook(),
+            c == null ? null : c.getNotebook());
+    if (gitConflict != null && notebookConflict != null) {
+
+      return Arrays.asList(
+          Utils.removeEmpty(new AutogenCodeDiff().setGit(gitConflict)),
+          Utils.removeEmpty(new AutogenCodeDiff().setNotebook(notebookConflict)));
+    } else {
+      return Collections.singletonList(
+          Utils.removeEmpty(
+              new AutogenCodeDiff().setGit(gitConflict).setNotebook(notebookConflict)));
```

Comment:
If `gitConflict` & `notebookConflict` i think this line should throw an error because proto does not accept the `null` value. just need to cross-check.

Reply:
that was the advantage of the Autogen classes.